### PR TITLE
Merge blazor-wasm in to master

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
https://github.com/dotnet/aspnetcore/pull/22296/commits/592dfe165b6fccb65accd20c3581a430b987127d has most of the master specific changes.

File: .config/dotnet-tools.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "isRoot": true,
   "tools": {
     "dotnet-serve": {
-      "version": "1.5.0",
+      "version": "1.7.125",
```
Comment (pranavkm) [inline .config/dotnet-tools.json:6]: This version supports assigning dotnet-serve a random port.

File: global.json [hunk 2]
```diff
@@ -11,6 +11,12 @@
       ],
       "dotnet/x86": [
         "$(MicrosoftNETCoreAppInternalPackageVersion)"
+      ],
+      "aspnetcore/x64": [
```
Comment (pranavkm) [inline global.json:15]: dotnet-serve doesn't roll forward to a 5.0 runtime.

File: src/ProjectTemplates/Shared/DevelopmentCertificate.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Certificates.Generation;
+
+namespace Templates.Test.Helpers
+{
+    public readonly struct DevelopmentCertificate
+    {
+        public DevelopmentCertificate(string certificatePath, string certificatePassword, string certificateThumbprint)
```
Comment (pranavkm) [inline src/ProjectTemplates/Shared/DevelopmentCertificate.cs:12]: FYI @javiercn

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/template.json [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+  "precedence": "6001",
+  "guids": [
+    "4C26868E-5E7C-458D-82E3-040509D0C71F",
+    "5990939C-7E7B-4CFA-86FF-44CA5756498A",
+    "650B3CE7-2E93-4CC4-9F46-466686815EAA",
+    "0AFFA7FD-4E37-4636-AB91-3753E746DB98",
+    "09732173-2cef-46b7-83db-1334bcb079d3", // Tenant ID
+    "53bc9b9d-9d6a-45d4-8429-2a2761773502" // Client ID
+  ],
+  "identity": "Microsoft.Web.Blazor.Wasm.CSharp.5.0",
```
Comment (pranavkm) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/template.json:22]: FYI @phenning

File: src/ProjectTemplates/Shared/DevelopmentCertificate.cs (hunk 3 above)
Comment (javiercn) [reply src/ProjectTemplates/Shared/DevelopmentCertificate.cs:12]: LGTM

File: eng/Baseline.xml [hunk 5]
```diff
@@ -36,6 +37,12 @@ Update this list when preparing for a new patch.
   <Package Id="Microsoft.AspNetCore.Components.Authorization" Version="3.1.4" />
   <Package Id="Microsoft.AspNetCore.Components.Forms" Version="3.1.4" />
   <Package Id="Microsoft.AspNetCore.Components.Web" Version="3.1.4" />
+  <Package Id="Microsoft.AspNetCore.Components.WebAssembly" Version="3.2.0" />
```
Comment (wtgodbe) [inline eng/Baseline.xml:40]: Can the 3.2.0-preview1 versions above be removed?

**Review by wtgodbe — APPROVED** [review]
Looks good other than the 1 question  (which might also apply to the 3.1 PR)

File: eng/Baseline.xml (hunk 5 above)
Comment (pranavkm) [reply eng/Baseline.xml:40]: Yes. Let me get this in and I'll follow up with a clean up

**Pull request merged**